host/src/containers/GameSessionContainer.tsx

```diff
@@ -104,20 +112,66 @@ const GameSessionContainer = () => {
     }
   }, [isModalOpen]);
 
+  useEffect(() => { //headerGame timer, 1 second interval, update localstorage in case page resets
+    if (gameTimer && !gameTimerZero){
+      let refreshIntervalId = setInterval(() => {
+        if (headerGameCurrentTime > 0) {
+          setHeaderGameCurrentTime(headerGameCurrentTime - 1);
+          localStorage.setItem('currentGameTimeStor', headerGameCurrentTime-1);
+        }
+        else
+          setGameTimerZero(true);
+      }, 1000);
+      return () => clearInterval(refreshIntervalId);
+    }
+  }, [gameTimer, gameTimerZero, headerGameCurrentTime]);
+
+  useEffect(()=>{ //update gameSession currentTimer every 3 seconds with time from headerGame Timer 
+    if (gameTimer && !gameTimerZero){
+      let refreshIntervalId = setInterval(() => {
+          let newUpdates = {currentTimer: (localStorage.getItem('currentGameTimeStor')>= 0 ? localStorage.getItem('currentGameTimeStor') : 0)};
+          apiClient.updateGameSession({ id: gameSessionId, ...newUpdates });
+      }, 3000);
+      return () => clearInterval(refreshIntervalId);
+    }
+  }, [gameTimer, gameTimerZero]);
+
   const handleUpdateGameSession = (newUpdates: Partial<IGameSession>) => {
     apiClient.updateGameSession({ id: gameSessionId, ...newUpdates })
       .then(response => {
+
+        if (response.currentState === stateArray[2])
+          setHeaderGameCurrentTime(response.phaseOneTime);
+        else if (response.currentState === stateArray[6])
+          setHeaderGameCurrentTime(response.phaseTwoTime);
+
         setGameSession(response);
+        checkGameTimer(response);
       });
   };
 
+  const checkGameTimer = (gameSession) =>{
```

Comment:
nit: Space after `=>`. There's a way to set the formatting in vscode so it formats for you on save. I don't think we've standardized it and we don't use a common pattern among apps. @lucahendicott might have the `vscode` setting that you can use to format the code for you that matches the pattern we use for host

Reply:
space fixed. those settings would be awesome though